chain/store/store.go

```diff
@@ -210,6 +211,24 @@ func (cs *ChainStore) SubscribeHeadChanges(f func(rev, app []*types.TipSet) erro
 	cs.headChangeNotifs = append(cs.headChangeNotifs, f)
 }
 
+func (cs *ChainStore) IsTipSetValidated(ctx context.Context, ts *types.TipSet) (bool, error) {
```

Comment:
I think @arajasek is right, and we should probably cache validation at a block level rather than at a tipset level.